src/v/datalake/coordinator/frontend.cc

```diff
@@ -255,7 +255,7 @@ frontend::add_translated_data_files_locally(
   ss::shard_id shard) {
     co_return co_await _coordinator_mgr->invoke_on(
       shard,
-      [&coordinator_partition,
+      [coordinator_partition,
```

Comment:
curious, is there a crash? or just want to avoid a cross shard reference?

Reply:
crash

Reply:
IMO the function should have been changed to take the partition by value, and then moved into the capture. the caller moves a stack allocated ntp and returns immediately (ie, without co_await)--that's a classic use after free scenario waiting ot happen.